### DISCUSSION TITLE
Downgrade rabbitmq-server from version 3.8.9 to 3.7.11

### DISCUSSION THREAD

**Discussion by anandtripathi5** [rabbitmq/rabbitmq-server#2735] [Other] [answered]
I want to downgrade the rabbitmq-server to a specific version and also the erlang cause the classic format of config is not supported in the newer version of rabbitmq-server. I'm looking for a solution for Ubuntu 18 version.

**Comment by michaelklishin** [comment]
I will convert this issue to a GitHub discussion. Currently GitHub will [automatically close and lock the issue](https://github.com/github/feedback/discussions/3197) even though your question will be transferred and responded to elsewhere. This is to let you know that we do not intend to ignore this but this is how the current GitHub conversion mechanism makes it seem for the users :(

**Comment by michaelklishin** [comment, marked as answer]
@anandtripathi5 RabbitMQ does not support in place downgrades. There are internal data store migrations, message store or Raft log format changes, and other kind of changes that are irreversible. You can use the [Blue/Green deployment strategy](https://www.rabbitmq.com/blue-green-upgrade.html) to move from one cluster to another.

However, I strongly believe that it would be **wrong** for you to downgrade. First of all, the classic config format [**is absolutely supported** by 3.8](https://www.rabbitmq.com/configure.html#configuration-files), from an additional `advanced.config` file to accompany the main `rabbitmq.conf` to the good old `rabbitmq.config` in the classic Erlang term config format. In fact, `rabbitmq.conf` is validated and then translated to the Erlang term format under the hood, so removing support for it would be impossible. If you have questions about using the classic config format with 3.8, you are welcome to ask them in separate discussions or on the mailing list or our community Slack.

Another reason why this is a bad idea is that [RabbitMQ 3.7 is entirely out of support](https://www.rabbitmq.com/versions.html). We will not produce any more 3.7 releases, including for serious issues and security issues. Needless to say you will be missing our one several [major new features](https://www.rabbitmq.com/changelog.html) only available in the `3.8.x` series.

**Comment by gerhard** [comment]
I'm curious as to why you want to downgrade to v3.7.11 @anandtripathi5?

**Reply by michaelklishin** [reply]
According to the original question, "cause the classic format of config is not supported in the newer version of rabbitmq-server" (which is not correct as I explained above)

**Reply by gerhard** [reply]
Is that the reason @anandtripathi5?

**Comment by anandtripathi5** [comment]
Actually, recently we stuck into an error "certify received CLIENT ALERT: Fatal - Unknown CA".

We are using 5671 port and amqps protocol. The configuration was fine and everything was working fine. But suddenly it stopped yesterday and this error was coming continuously. Now we have used 5672 port(without SSL) and meanwhile just debugging why it failed for SSL.

In the meantime, I thought maybe version 3.8.9 is giving error cause on our prod environments everything is working fine on 3.7.11 that's why I initiated this discussion. But I guess we can close this cause the version is nothing to do with my current problem. Thanks @michaelklishin for the quick reply.

**Reply by michaelklishin** [reply]
"Unknown CA" is explicitly mentioned in the TLS guide's section on [peer verification](https://www.rabbitmq.com/ssl.html#peer-verification-how-it-works). It is absolutely not version- or configuration format-specific. Your client certificate is ultimately signed by a CA that's not trusted by the RabbitMQ node it has connected to.
